app/controllers/api/episodes_controller.rb

```diff
@@ -56,8 +56,8 @@ def visible?
     visible
   end
 
-  def included(relation)
-    relation.with_deleted
+  def find_base
```

Comment:
This gets a bit into hal_api-rails internals ...

But apparently in the 0.3.x upgrade, `included` now gets called on index resources as well as show.  So to target the `with_deleted` only at show resources (so we get 410s instead of 404s), i moved to the `find_base` override.

This was the best place I could find to differentiate between the index-vs-show query.  The only other option I could see would be setting `@episode` myself ... but that feels less clean.  But maybe I missed something.

Reply:
yeah, this makes sense, it does make me wonder if we need some better way for hal_api-rails to work for this.